tests-web/www/js/main.js

```diff
@@ -1,7 +1,230 @@
-import { makeDropHandler, makeLinkHandler, handleDragOver } from "./files.js";
-import { makeImportCurrentView } from "./import.js";
-import { makeOpenTest, loadTests } from "./tests.js";
+import {
+  loadFile,
+  makeDropHandler,
+  makeLinkHandler,
+  handleDragOver,
+  downloadGeneratedFile,
+} from "./files.js";
+import { loadTests } from "./tests.js";
+import {
+  makeOsmLayer,
+  makePlainGeoJsonLayer,
+  makeDetailedGeoJsonLayer,
+  makeDotLayer,
+} from "./layers.js";
+import init, { JsStreetNetwork } from "./osm2streets-js/osm2streets_js.js";
 
+await init();
+
+export class StreetExplorer {
+  constructor(mapContainer) {
+    this.map = setupLeafletMap(mapContainer);
+    this.layerControl = L.control.layers({}, {}).addTo(this.map);
+    this.currentTest = null;
+
+    // Add all tests to the sidebar
+    loadTests();
+  }
+
+  static async create(mapContainer) {
+    const app = new StreetExplorer(mapContainer);
+
+    // Possibly load a test from the URL
+    const q = new URLSearchParams(window.location.search);
+    if (q.has("test")) {
+      const test = q.get("test");
+      console.info(`Loading test ${test} from URL.`);
+      await app.setCurrentTest((app) => TestCase.loadFromServer(app, test));
+    }
+
+    // Wire up the import button
+    const importButton = document.getElementById("import-view");
+    importButton.onclick = async () => {
+      if (app.map.getZoom() < 15) {
+        window.alert("Zoom in more to import");
+        return;
+      }
+
+      await app.setCurrentTest((app) =>
+        TestCase.importCurrentView(app, importButton)
+      );
+    };
+
+    return app;
+  }
+
+  addLayer(name, layer) {
+    layer.addTo(this.map);
+    this.layerControl.addOverlay(layer, name);
+    return layer;
+  }
+
+  removeLayer(layer) {
+    this.map.removeLayer(layer);
+    this.layerControl.removeLayer(layer);
+  }
+
+  async setCurrentTest(testMaker) {
+    this.currentTest?.cleanup();
+    this.currentTest = await testMaker(this);
+    this.map.fitBounds(this.currentTest.bounds, { animate: false });
```

Comment:
I'm rethinking the error handling now... maybe `testMaker` should be allowed to throw an exception. We catch it here, just clear out the old layer but do nothing else. Would that be cleaner?